From e89f577e5a0530197f3d0bff2b196052ffd0be38 Mon Sep 17 00:00:00 2001
From: zou cao <cao.zou@windriver.com>
Date: Mon, 20 Feb 2017 10:04:20 +0800
Subject: [PATCH] arm: dts: zynq: add support for picozed

Create a board-related dts for picozed dependency on zynq-7000.dtsi

Signed-off-by: zou cao <cao.zou@windriver.com>
---
 arch/arm/boot/dts/zynq-picozed.dts |  129 ++++++++++++++++++++++++++++++++++++
 1 files changed, 129 insertions(+), 0 deletions(-)
 create mode 100644 arch/arm/boot/dts/zynq-picozed.dts

diff --git a/arch/arm/boot/dts/zynq-picozed.dts b/arch/arm/boot/dts/zynq-picozed.dts
new file mode 100644
index 0000000..d8bd2c5a
--- /dev/null
+++ b/arch/arm/boot/dts/zynq-picozed.dts
@@ -0,0 +1,129 @@
+/*
+ *  Copyright (C) 2015 Xilinx
+ *  Copyright (C) 2015 WindRiver Corp.
+ *
+ * This software is licensed under the terms of the GNU General Public
+ * License version 2, as published by the Free Software Foundation, and
+ * may be copied, distributed, and modified under those terms.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/dts-v1/;
+/include/ "zynq-7000.dtsi"
+
+/ {
+	model = "Avnet picoZed";
+	compatible = "xlnx,zynq-picozed", "xlnx,zynq-7000";
+
+	aliases {
+		ethernet0 = &gem0;
+		i2c0 = &i2c0;
+		serial0 = &uart1;
+		spi0 = &qspi;
+		mmc0 = &sdhci0;
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0x0 0x40000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyPS0,115200 root=/dev/ram rw earlyprintk";
+		linux,stdout-path = &uart1;
+		stdout-path = &uart1;
+	};
+
+	amba@0 {
+		/* Disabled Devices */
+		ps7_sd_0: sdio@e0100000 { compatible = "invalid"; };
+		ps7_uart_0: serial@e0000000 { compatible = "invalid"; };
+		ps7_ethernet_1: ethernet@e000c000 { compatible = "invalid"; };
+		ps7_i2c_0: i2c@e0004000 { compatible = "invalid"; };
+		ps7_i2c_1: i2c@e0005000 { compatible = "invalid"; };
+		ps7_wdt_0: wdt@f8005000 { compatible = "invalid"; };
+		ps7_can_0: can@e0008000 { compatible = "invalid"; };
+		ps7_can_1: can@e0009000 { compatible = "invalid"; };
+		ps7_usb_1: usb@e0003000 { compatible = "invalid"; };
+	};
+
+	usb_phy0: phy0 {
+		compatible = "ulpi-phy";
+		phy-cells = <0>;
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
+
+};
+
+&qspi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "micron,m25p80", "spansion,s25fl128s", "jedec,spi-nor";
+		reg = <0x0>;
+		spi-max-frequency = <50000000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		partition@qspi-fsbl-uboot {
+			label = "qspi-fsbl-uboot";
+			reg = <0x0 0x100000>;
+		};
+		partition@qspi-linux {
+			label = "qspi-linux";
+			reg = <0x100000 0x500000>;
+		};
+		partition@qspi-device-tree {
+			label = "qspi-device-tree";
+			reg = <0x600000 0x20000>;
+		};
+		partition@qspi-rootfs {
+			label = "qspi-rootfs";
+			reg = <0x620000 0x5E0000>;
+		};
+		partition@qspi-bitstream {
+			label = "qspi-bitstream";
+			reg = <0xC00000 0x400000>;
+		};
+	};
+} ;
+
+&usb0 {
+	status = "okay";
+	usb-phy = <&usb_phy0>;
+};
+
+&sdhci1 {
+	status = "okay";
+};
+
+&sdhci0 {
+	status = "okay";
+};
+
+&uart1 {
+	status = "okay";
+};
+
+&watchdog0 {
+	status = "okay";
+	reset-on-timeout;
+};
+
+&gem0 {
+	compatible = "xlnx,ps7-ethernet-1.00.a";
+	status = "okay";
+	clocks = <&clkc 30>, <&clkc 13>;
+	clock-names =  "aper_clk", "ref_clk";
+	phy-handle = <&ethernet_phy>;
+	phy-mode = "rgmii-id";
+
+
+	ethernet_phy: ethernet-phy@0 {
+		reg = <0>;
+	} ;
+};
-- 
1.7.5.4

